src/detect-tcp-window.c

```diff
@@ -226,87 +226,66 @@ void DetectWindowFree(DetectEngineCtx *de_ctx, void *ptr)
  */
 static int DetectWindowTestParse01 (void)
 {
-    int result = 0;
     DetectWindowData *wd = NULL;
     wd = DetectWindowParse(NULL, "35402");
-    if (wd != NULL &&wd->size==35402) {
-        DetectWindowFree(NULL, wd);
-        result = 1;
-    }
+    FAIL_IF_NOT(wd != NULL &&wd->size==35402);
```

Comment:
Could we split this into two checks? 
This makes it easier to verify which check is failing if this test fails.

Reply:
Understood, I'll make the edit